recipes/protobuf/all/conanfile.py

```diff
@@ -59,17 +67,40 @@ def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
 
+        if self._protobuf_release < "27.0":
+            self.options.rm_safe("upb")
+
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_zlib:
             self.requires("zlib/[>=1.2.11 <2]")
 
+        if self._protobuf_release >= "22.0":
+            self.requires("abseil/20240116.2", transitive_headers=True)
+
     def validate(self):
         if self.options.shared and is_msvc_static_runtime(self):
             raise ConanInvalidConfiguration("Protobuf can't be built with shared + MT(d) runtimes")
-
+        
+        if is_msvc(self) and self._protobuf_release >= "22" and self.options.shared and \
+            not self.dependencies["abseil"].options.shared:
```

Comment:
Is too risk enforce `self.options["abseil"].shared = True`? In Conan 1.x, we will not have packages for msvc + shared due mixed profiles.

Reply:
Don't think it's worth it particularly - we know that shared works because of the Conan 2 checks, and in other recent cases we already decided to go the `validate()` way (and point the user to a more sensible choice), than to embed static inside shared (although on Windows it may be less of a risk).

Reply:
> in other recent cases we already decided to go the validate() way (and point the user to a more sensible choice), than to embed static inside shared (although on Windows it may be less of a risk).

Good to know. Thanks!